File: Meaium/AddTwoNumbers.js
"Link: https://leetcode.com/problems/add-two-numbers/"
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
 var addTwoNumbers = function(l1, l2) {
    let sum = readNode(l1) + readNode(l2);
    let reversedSum = String(sum).split("").reverse().join("");
    
    let result = new ListNode();
    result.val = reversedSum[0];
    reversedSum = reversedSum.substring(1);
    
    for (i of reversedSum) {
        insertNode(result, i);
    }
    return result;
};

function insertNode(node, val) {
    let point = node;
    while (point.next !== null) {
        point = point.next;
    }
    point.next = new ListNode();
    point.next.val = val;
    return node;
}

function arrayToNumber(array) {
    let result = "";
    for (let i of array) {
        result += i;
    }
    return BigInt(result);
}

function readNode(node) {
    let result = [];
    while (node !== null) {
        result.push(node.val);
        node = node.next;
    }
    result = arrayToNumber(result.reverse());
    return result;
}

File: Add-Two-Numbers/Add-Two-Numbers.cpp
/**
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* dummy=new ListNode();
        ListNode* temp=dummy;
        int carry=0;
        while(l1!=NULL || l2!=NULL || carry){
            int sum=0;
            if(l1!=NULL){
                sum+=l1-&gt;val;
                l1=l1-&gt;next;
            }
            if(l2!=NULL){
                sum+=l2-&gt;val;
                l2=l2-&gt;next;
            }
            sum+=carry;
            carry=sum/10;
            ListNode* newnode=new ListNode(sum%10);
            temp-&gt;next=newnode;
            temp=temp-&gt;next;
        }
        return dummy-&gt;next;
    }
};